src/components/Footer.tsx

```diff
@@ -0,0 +1,94 @@
+import classNames from 'classnames';
+import React, { useMemo, useState } from 'react';
+import { Todo } from '../types/Todo';
+import { Filter } from '../types/Filter';
+
+type Props = {
+  todos: Todo[],
+  setFilter: React.Dispatch<React.SetStateAction<Filter>>,
+};
+
+export const Footer: React.FC<Props> = ({ todos, setFilter }) => {
+  const [selectedLink, setSelectedLink] = useState<string | null>('All');
+
+  const handleAllChange = (event: React.MouseEvent<HTMLAnchorElement>) => {
+    event.preventDefault();
+    const selectedLinkText = event.currentTarget.textContent;
+
+    setSelectedLink(selectedLinkText);
+    setFilter(Filter.All);
+  };
```

Comment:
the function can be created as common for all
```suggestion
  const handleChange = (filterType: Filter) => (event: React.MouseEvent<HTMLAnchorElement>) => {
    event.preventDefault();
    const selectedLinkText = event.currentTarget.textContent;

    setSelectedLink(selectedLinkText);
    setFilter(filterType);
  };
```